src/main/kotlin/odyseja/odysejapka/tournamentManager/TournamentManagerController.kt

```diff
@@ -0,0 +1,26 @@
+package odyseja.odysejapka.tournamentManager
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+data class ZspIdRequest(
+        val zspId: String
+)
+@RestController
+@RequestMapping("/api/v1/tm")
+class TournamentManagerController(private val tournamentManagerService: TournamentManagerService) {
+    @PostMapping("/generate")
+    fun getCsv(@RequestParam zspId: String): ResponseEntity<ByteArray> {
+
+        val csvData = tournamentManagerService.generateCsv(zspId)
+
+        return ResponseEntity
+                .ok()
+                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=your_file.csv")
```

Comment:
`your_file.csv` to chyba jakaś placeholderowa nazwa :P

Reply:
tbh nawet nw do końca czy to ma znaczenie czy jest wgl